collector/src/main/java/io/prometheus/jmx/JmxCollector.java

```diff
@@ -69,13 +69,21 @@
 
     private static final Pattern snakeCasePattern = Pattern.compile("([a-z0-9])([A-Z])");
 
+    private final JmxMBeanPropertyCache jmxMBeanPropertyCache;
+
+    private JmxCollector() {
+        jmxMBeanPropertyCache = new JmxMBeanPropertyCache();
```

Comment:
Good point, I'll move it there. Out of habit, I initialize in constructor so that I can more easily mock objects out. In this case it is just a map behind the scenes, so doubt we'd need to mock anything.